app/src/main/java/com/mapbox/navigation/examples/alternative/ShowAlternativeRoutesActivity.kt

```diff
@@ -0,0 +1,405 @@
+                .accessToken(getString(R.string.mapbox_access_token))
+                .locationEngine(ReplayLocationEngine(mapboxReplayer))
+                .routeAlternativesOptions(
+                    RouteAlternativesOptions.Builder()
+                        .intervalMillis(TimeUnit.SECONDS.toMillis(30))
+                        .build()
+                )
+                .build()
+        )
+        initNavigation()
+        initStyle()
+        initListeners()
+    }
+
+    override fun onStart() {
+        super.onStart()
+        mapboxNavigation.registerLocationObserver(locationObserver)
+        mapboxNavigation.registerRouteProgressObserver(replayProgressObserver)
+        mapboxNavigation.registerRouteProgressObserver(routeProgressObserver)
+        mapboxNavigation.registerRoutesObserver(routesObserver)
+        mapboxNavigation.registerRouteAlternativesObserver(alternativesObserver)
+    }
+
+    override fun onStop() {
+        super.onStop()
+        mapboxNavigation.unregisterRouteProgressObserver(replayProgressObserver)
+        mapboxNavigation.unregisterRouteProgressObserver(routeProgressObserver)
+        mapboxNavigation.unregisterLocationObserver(locationObserver)
+        mapboxNavigation.unregisterRoutesObserver(routesObserver)
+        mapboxNavigation.unregisterRouteAlternativesObserver(alternativesObserver)
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        routeLineApi.cancel()
+        routeLineView.cancel()
+        mapboxReplayer.finish()
+        mapboxNavigation.onDestroy()
+    }
+
+    @SuppressLint("MissingPermission")
+    private fun initListeners() {
+        binding.startNavigation.setOnClickListener {
+            mapboxNavigation.startTripSession()
+            binding.startNavigation.visibility = View.GONE
+            findRoute(originPoint, destinationPoint)
+        }
+
+        binding.mapView.gestures.addOnMapClickListener(mapClickListener)
+    }
+
+    /**
+     * Request routes between the two points.
+     */
+    private fun findRoute(origin: Point?, destination: Point?) {
+        val routeOptions = RouteOptions.builder()
+            .applyDefaultNavigationOptions()
+            .applyLanguageAndVoiceUnitOptions(this)
+            .coordinatesList(listOf(origin, destination))
+            .layersList(listOf(mapboxNavigation.getZLevel(), null))
+            // set this flag to true if you want to request alternative routes as well
+            .alternatives(true)
+            .build()
+        mapboxNavigation.requestRoutes(
+            routeOptions,
+            object : NavigationRouterCallback {
+                override fun onRoutesReady(
+                    routes: List<NavigationRoute>,
+                    routerOrigin: RouterOrigin
+                ) {
+                    if (routes.isNotEmpty()) {
+                        mapboxNavigation.setNavigationRoutes(routes)
+                        startSimulation(routes.first())
```

Comment:
I removed this line and nothing changed. It seems that `ReplayProgressObserver` can do the work alone, can't it?

Reply:
Idk, I removed this line and the simulation does not work at all...

Reply:
interesting 🤔 
let me try one more time

Reply:
Tried. it works for me :)